src/QsCompiler/DataStructures/Diagnostics.fs

```diff
@@ -434,6 +463,11 @@ type DiagnosticItem =
         let ApplyArguments =
             DiagnosticItem.ApplyArguments args
             << function
+            | ErrorCode.TypeMismatch ->
+                "The type {1} does not match the type {0}.\nActual type:   {3}\nExpected type: {2}"
+            | ErrorCode.MissingBaseType ->
+                "The type {1} does not {0} the type {2}.\nLeft-hand type:  {3}\nRight-hand type: {4}"
```

Comment:
What is the argument 0 here?

Reply:
A description of the relationship between the two types. Possible values are "share a subtype with", "equal", or "share a base type with". I think maybe the error name `MissingBaseType` should be renamed to something like `MissingIntersection`.

Reply:
Renamed to `TypeIntersectionMismatch`.

Reply:
Thanks.